### DISCUSSION TITLE
Is there a way to access useTheme hook in _app.js

### DISCUSSION THREAD

**Discussion by junaid33** [shuding/nextra#490] [Q&A] [answered]
I'm trying to use Mantine with Nextra and am trying to get dark mode to sync between the 2. I try to get the theme using useTheme from the next-themes package, but it's not working in _app.js. I can access it in the component directly but now have to render the MantineProvider for each component.

**Comment by promer94** [comment, marked as answer]
First you need to install the latest version of next-themes and nextra
```json
"dependencies": {
    "next-themes": "0.2.0-beta.2",
    "nextra-theme-docs": "2.0.0-alpha.54",
    "nextra": "2.0.0-alpha.54"
    ....
  }
```

Then you should ba able to access the theme in _app.js
```jsx
import 'nextra-theme-docs/style.css'
import { useTheme } from 'next-themes'
import React from 'react'

const TestTheme = ({ children }) => {
  const { theme, setTheme, systemTheme } = useTheme()
  console.log('theme', theme, systemTheme)
  return children
}

export default function MyApp({ Component, pageProps }) {
  // Use the layout defined at the page level, if available
  const getLayout = Component.getLayout || (page => page)
  return getLayout(
    <TestTheme>
      <Component {...pageProps} />
    </TestTheme>
  )
}

```


Or you can find the theme from localStoreage 
<img width="1153" alt="Screen Shot 2022-06-22 at 22 48 37" src="https://user-images.githubusercontent.com/15878786/175059567-bd190a99-5789-453e-89de-018068a3ee37.png">

**Reply by junaid33** [reply]
I tried this, but it's showing theme undefined undefined. Here's a reproduction: https://github.com/junaid33/nextra-theme-repro. 

I also tried getting the theme value from local storage and it works, but when the local storage value is changed, it doesn't update. Looks like Nextra saves the theme to state and then updates local storage. Anyway to access that state directly?

**Reply by promer94** [reply]
# Problem
yarn seems to install multiple next-themes in node_modules 
<img width="900" alt="Screen Shot 2022-06-23 at 03 34 00" src="https://user-images.githubusercontent.com/15878786/175122196-27c9aa4e-8229-4818-ab87-c950b8c5b105.png">

npm (pnpm) will only install a signle next-themes 
<img width="667" alt="Screen Shot 2022-06-23 at 03 38 15" src="https://user-images.githubusercontent.com/15878786/175122341-a1a56aaf-9985-4642-8a45-df97cd22dc68.png">

# Workaround
For yarn, need to remove next-themes from dependencies
```json
"dependencies": {
    "nextra-theme-docs": "2.0.0-alpha.54",
    "nextra": "2.0.0-alpha.54"
}
```

Thanks for your reproduction, it's really helpful

**Reply by junaid33** [reply]
It worked! Thanks for your help!

**Reply by junaid33** [reply]
This is working great, but there is one bug. When I go from system to dark mode and refresh, the Mantine components go to light mode. You can try it here: https://docs.openship.org and the _app.js: https://github.com/openshiporg/docs/blob/main/pages/_app.js.

**Reply by whatsoevan** [reply]
I tried this answer out in the blog theme, and useTheme still produces "theme undefined undefined" in the console.


```
export default function Nextra({ Component, pageProps }) {
  const { theme, setTheme, systemTheme } = useTheme()
  console.log('theme', theme, systemTheme)
  return (
    <>
      <Head>
        <link
          rel="alternate"
          type="application/rss+xml"
          title="RSS"
          href="/feed.xml"
        />
        <link
          rel="preload"
          href="/fonts/Inter-roman.latin.var.woff2"
          as="font"
          type="font/woff2"
          crossOrigin="anonymous"
        />
      </Head>
      <GeistProvider>
        <Component {...pageProps} />
      </GeistProvider>
    </>
  )
}

```

Here is the path in my node_modules to the nextra-theme-blog.

C:\Users\{username}\project\node_modules\nextra-theme-blog\node_modules\next-themes
In the blog theme, there appears to be another next themes as well. 

There is next-themes, nextra, and nextra-theme-blog in my node_modules.

Am I supposed to remove next-themes from my dependencies? 
Or the one inside node_modules?

**Reply by TheMikeyRoss** [reply]
Did you find a solution for this? @evanxshu

**Reply by arno-fukuda** [reply]
>This is working great, but there is one bug. When I go from system to dark mode and refresh, the Mantine components go to light mode. You can try it here: https://docs.openship.org/ and the _app.js: https://github.com/openshiporg/docs/blob/main/pages/_app.js.

Having the same issue.

Anyone has a solution?

**Reply by JanBrin** [reply]
@arno-fukuda A workaround is creating a new hook that sets a "isDarkMode" variable in my case:

```typescript
    const { resolvedTheme } = useTheme();
    const [isDarkMode, setIsDarkMode] = useState(false);

    useEffect(() => {
        setIsDarkMode(resolvedTheme === 'dark');
    }, [resolvedTheme]);
```

then using "isDarkMode" in the code to check what theme is used.
